pkgs/development/compilers/zig/bootstrap.nix

```diff
@@ -0,0 +1,44 @@
+{
+  lib,
+  stdenvNoCC,
+  callPackage,
+  targetPlatform,
+  version,
+  bootstrapHashes ? { },
+}:
+stdenvNoCC.mkDerivation (
+  finalAttrs:
+  let
+    common = callPackage ./common.nix {
+      inherit (finalAttrs) pname version;
+      zig = finalAttrs.finalPackage;
+    };
+
+    platform = "${
+      if targetPlatform.isDarwin then "macos" else targetPlatform.parsed.kernel.name
+    }-${targetPlatform.parsed.cpu.name}";
+  in
+  {
+    pname = "zig-bootstrap";
+    inherit version;
+
+    bootstrap = builtins.fetchTarball {
```

Comment:
Is it okay to use fetchTarball here? We generally do not do that in Nixpkgs, even for bootstrap tarballs. (Instead we use <nix/fetchurl.nix>)

Reply:
I'm not sure, doesn't `fetchTarball` unpack? Does `fetchurl` boot unpack?

Reply:
It must do, or we wouldn't be able to unpack the bootstrap tarballs.

Reply:
I'm not sure I actually understand what the purpose of the bootstrap Zig is, actually?  Why didn't we need it before, but we do now?

Reply:
It's a prebuilt version of Zig and I'm wanting to be able to have a bootstrap stdenv (`pkgsArocc` / `pkgsZig`) but without the cross stdenv since I want to try bootstrapping LLVM that way.

Reply:
In that case, wouldn't it make more sense to build our own bootstrap binaries in the normal way, rather than downloading a binary from elsewhere?

Reply:
Idk, it's mostly experimental stuff until it can actually be worked out.

Reply:
I think it needs to be a bit more worked out than this before it goes in, because it's difficult to evaluate currently whether this is actually a positive change.  If we're adding a new package set in a way that's different from how we'd normally do it (get it working with cross first, cross compile the bootstrap binaries, then upload them to tarballs.nixos.org), that needs to be justified.

Reply:
Hmm, so we drop the bootstrap stuff and save it for a later PR?

Reply:
I dropped the bootstrap stuff.